### ISSUE TITLE
VGG.Training_ShouldReduceLoss: fused-compiled training engages on step 1 then refuses on step 2 — "plan-embedded Adam state cannot be transferred"

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1395]
## Summary

\`VGGNetworkTests.Training_ShouldReduceLoss\` fails on current master with:

\`\`\`
System.InvalidOperationException : Fused compiled training has already run successfully,
but the current step cannot engage the fused path. The plan-embedded Adam/AdamW/SGD state
cannot be transferred to the eager optimizer, so falling back silently would produce a
trajectory that diverges from the previous fused steps. Common causes: variable input/target
shape (new compiled plan), LR scheduler or adaptive-rate changes, attached AMSGrad.
Resolution: keep shapes and optimizer hyperparameters stable across steps, OR call
ResetState() / InvalidateParameterCountCache() to explicitly reset training state, OR disable
compilation (AiModelBuilder.ConfigureJitCompilation(JitCompilationConfig.Disabled)).
\`\`\`

at \`NeuralNetworkBase.TryTrainWithFusedOptimizer:6240\`. Test execution time: ~53 s of the 120 s timeout — so step 1 ran the fused path successfully and step 2 (next iteration of the test's same-input-same-target loop) hit the throw.

## What the test does

\`\`\`csharp
[Fact(Timeout = 120000)]
public virtual async Task Training_ShouldReduceLoss()
{
    var rng = ModelTestHelpers.CreateSeededRandom();
    using var network = CreateNetwork();      // VGGNetwork<double>() default ctor
    var input  = CreateRandomTensor([1, 3, 224, 224], rng);
    var target = CreateRandomTargetTensor([1000], rng);

    var initialOutput = network.Predict(input);
    double initialLoss = ComputeMSE(initialOutput, target);

    for (int i = 0; i < TrainingIterations * 3; i++)   // 30 iters by default
        network.Train(input, target);

    var finalOutput = network.Predict(input);
    double finalLoss = ComputeMSE(finalOutput, target);

    // assert finalLoss <= initialLoss + tol
}
\`\`\`

No LR scheduler. No AMSGrad. No batch-size change. Same input + target every step. The \"common causes\" listed in the error message all rule out — yet the second Train still can't engage the fused path.

## Why this isn't \"obvious\" causes

Investigation in PR session for [#1306](https://github.com/ooples/AiDotNet/issues/1306):

- **Plan-cache key**: \`CompiledTapeTrainingStep.TryStepWithFusedOptimizer\` builds a composite key from \`input._shape\` + sentinel + \`target._shape\`. Same input shape and same target shape across calls → same key → \`cache.GetOrCompileTraining\` cache hit → same plan reference. \`ReferenceEquals(_configuredPlan, plan)\` at \`CompiledTapeTrainingStep:474\` should pass.
- **Optimizer config**: \`TryMapToFusedOptimizerConfig\` builds \`(OptimizerType, lr, β1, β2, ε, wd)\` from the live Adam options every call. No LR scheduler is attached, \`UseAdaptiveLearningRate\` defaults to false, \`UseAMSGrad\` defaults to false. \`learningRate = (float)adam.GetCurrentLearningRate()\` reads the stored \`_currentLearningRate\` which is set once at Adam ctor and never mutated without a scheduler. \`currentConfig.Equals(_configuredOptimizerConfig)\` at \`CompiledTapeTrainingStep:482-489\` should pass.
- **Static fields are [ThreadStatic]**: \`_cache\`, \`_cachedParameters\`, \`_configuredPlan\`, \`_configuredOptimizerConfig\`, \`_persistentInput\`, \`_persistentTarget\` are all per-thread, so cross-test parallel races would NOT touch them (the same thread runs both Train calls inside one test method).
- **Predict-before-Train doesn't invalidate**: VGG.Predict calls \`SetTrainingMode(false)\`, runs Forward, restores mode. No \`InvalidateParameterCountCache\` call. No fused-state mutation.

So none of the documented common-causes apply, yet step 2 returns \`ran=false\` from \`TryStepWithFusedOptimizer\`.

## Next step

Drop into a minimal repro and instrument the three return-false paths in \`CompiledTapeTrainingStep.TryStepWithFusedOptimizer\` (\"plan switch\" at line 480 / \"config drift\" at line 488 / catch at line 513) to find which one fires on step 2. The catch path is most suspect — an exception swallowed in trace mode would produce exactly this symptom and the trace warning would name the underlying cause.

## Workaround for users

Per the error message: call \`ResetState()\` between Train steps (defeats fused-state retention but unblocks training), or set \`AiModelBuilder.ConfigureJitCompilation(JitCompilationConfig.Disabled)\` to take the entirely-eager path. Both are correctness-preserving stopgaps.

## NOT this issue

The 3 timeouts (\`ResNetNetworkTests.MoreData_ShouldNotDegrade\`, \`VGGNetworkTests.TrainingError_ShouldNotExceedTestError\`, \`VGGNetworkTests.LossStrictlyDecreasesOnMemorizationTask\`) are perf-gap failures filed as [#1394](https://github.com/ooples/AiDotNet/issues/1394). Different root cause.
